index.js

```diff
@@ -700,108 +738,133 @@ HardSourceWebpackPlugin.prototype.apply = function(compiler) {
       }
       return serialized;
     }
-
-    compilation.modules.forEach(function(module) {
-      var identifierPrefix = cachePrefix(compilation);
-      if (identifierPrefix === null) {
-        return;
-      }
-      var identifier = identifierPrefix + module.identifier();
-      var existingCacheItem = moduleCache[identifier];
-
-      if (
-        module.request &&
-        module.cacheable &&
-        !(module instanceof HardModule) &&
-        (module instanceof NormalModule) &&
-        (
-          existingCacheItem &&
-          module.buildTimestamp > existingCacheItem.buildTimestamp ||
-          !existingCacheItem
-        )
-      ) {
-        var source = module.source(
-          compilation.dependencyTemplates,
-          compilation.moduleTemplate.outputOptions,
-          compilation.moduleTemplate.requestShortener
-        );
-        var assets = Object.keys(module.assets || {}).map(function(key) {
-          return {
-            key: requestHash(key),
-            value: module.assets[key].source(),
-          };
+    var promises = dataCache.fileDependencies.map(function(file) {
+      if (fileMd5s[file]) {
+        return Promise.resolve();
+      } else {
+        return md5(file, compiler.inputFileSystem)
```

Comment:
it seems like I have to do this both https://github.com/mzgoddard/hard-source-webpack-plugin/pull/45/files#diff-168726dbe96b3ce427e7fedce31bb0bcR462 and here. 

In the case where no dataCache exists previously, or if a new file was added, that code path would never get called so I need to make sure to record any new values at this point in time, right before writing to the cache.

Let me know if you have any better ways in mind of doing this!